### DISCUSSION TITLE
scv.tl.velocity_graph(adata) ERROR

### DISCUSSION THREAD

**Discussion by TamaraSall** [theislab/scvelo#1085] [Q&A] [answered]
Hi,
I am running scv.tl.velocity_graph(adata) after computing velocities but it is giving me the below error. Please help me figure it out. I really appreciate your help!
:scv.tl.velocity
scv.tl.velocity(adata, mode='stochastic')
computing velocities
    finished (0:00:20) --> added 
    'velocity', velocity vectors for each individual cell (adata.layers)

:adata.var['velocity_genes'].sum()
1847

:adata.layers['velocity']
array([[ 0.00000000e+00,  1.01930305e-01,  0.00000000e+00, ...,
         0.00000000e+00,  5.75187922e-01, -8.72176979e-03],
       [ 0.00000000e+00,  2.69109141e-02,  0.00000000e+00, ...,
         0.00000000e+00,  3.71174335e-01,  7.54352659e-02],
       [ 0.00000000e+00,  4.26745087e-01,  1.88481901e-02, ...,
         0.00000000e+00,  1.14652634e-01,  2.53249910e-02],
       ...,
       [-1.17988602e-04,  6.98574245e-01,  2.98842527e-02, ...,
         0.00000000e+00,  4.40558761e-01,  3.17439176e-02],
       [-3.95343173e-04,  2.53678001e-02,  1.38029069e-01, ...,
         0.00000000e+00,  8.02107602e-02,  1.71904862e-02],
       [-4.73634282e-04,  2.10665725e-02, -2.74910722e-02, ...,
         0.00000000e+00,  1.14744556e+00, -3.20626167e-03]], dtype=float32)

:scv.tl.velocity_graph(adata)
computing velocity graph (using 1/24 cores)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[388], line 1
----> 1 scv.tl.velocity_graph(adata)

File ~/.conda/envs/scv/lib/python3.8/site-packages/scvelo/tools/velocity_graph.py:363, in velocity_graph(data, vkey, xkey, tkey, basis, n_neighbors, n_recurse_neighbors, random_neighbors_at_max, sqrt_transform, variance_stabilization, gene_subset, compute_uncertainties, approx, mode_neighbors, copy, n_jobs, backend)
    359 n_jobs = get_n_jobs(n_jobs=n_jobs)
    360 logg.info(
    361     f"computing velocity graph (using {n_jobs}/{os.cpu_count()} cores)", r=True
    362 )
--> 363 vgraph.compute_cosines(n_jobs=n_jobs, backend=backend)
    365 adata.uns[f"{vkey}_graph"] = vgraph.graph
    366 adata.uns[f"{vkey}_graph_neg"] = vgraph.graph_neg

File ~/.conda/envs/scv/lib/python3.8/site-packages/scvelo/tools/velocity_graph.py:175, in VelocityGraph.compute_cosines(self, n_jobs, backend)
    172 n_obs = self.X.shape[0]
    174 # TODO: Use batches and vectorize calculation of dX in self._calculate_cosines
--> 175 res = parallelize(
    176     self._compute_cosines,
    177     range(self.X.shape[0]),
    178     n_jobs=n_jobs,
    179     unit="cells",
    180     backend=backend,
    181 )()
    182 uncertainties, vals, rows, cols = map(_flatten, zip(*res))
    184 vals = np.hstack(vals)

File ~/.conda/envs/scv/lib/python3.8/site-packages/scvelo/core/_parallelize.py:138, in parallelize.<locals>.wrapper(*args, **kwargs)
    126     pbar, queue, thread = None, None, None
    128 res = Parallel(n_jobs=n_jobs, backend=backend)(
    129     delayed(callback)(
    130         *((i, cs) if use_ixs else (cs,)),
   (...)
    135     for i, cs in enumerate(collections)
    136 )
--> 138 res = np.array(res) if as_array else res
    139 if thread is not None:
    140     thread.join()

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (1, 4) + inhomogeneous part.

**Comment by WeilerP** [comment, marked as answer]
Fixed in #1059.
